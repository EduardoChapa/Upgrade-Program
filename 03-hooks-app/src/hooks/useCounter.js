import { useState } from 'react';

function useCounter(initialState = 0) {

    const [counter, setCounter] = useState(initialState);

    const increment = () => { setCounter(counter + 1); }

    const decrement = () => { setCounter(counter - 1); }

    const reset = () => { setCounter(0) }

    return {
        counter,
        increment,
        decrement,
        reset
    };
}

export default useCounter;