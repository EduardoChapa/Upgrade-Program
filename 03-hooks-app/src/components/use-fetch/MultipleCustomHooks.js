import React from 'react';
import useCounter from '../../hooks/useCounter';
import useFetch from '../../hooks/useFetch';

function MultipleCustomHooks() {

    const { counter, increment } = useCounter(1);

    const { data, loading } = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`);

    const { quote, author } = !!data && data[0];

    return(
        <>
            <h1>Funny Quotes</h1>
            <hr/>
            {
                loading ?
                (
                    <div className='alert alert-info text-center'>
                        Loading...
                    </div>
                ) :
                (
                    <blockquote className='blockquote text-left'>
                        <p className='mb-0'>{ quote }</p>
                        <br/>
                        <footer className='blockquote-footer'>{ author }</footer>
                    </blockquote>
                )
            }
            {/* <blockquote className='blockquote text-left'>
                <p className='mb-0'>{ quote }</p>
                <br/>
                <footer className='blockquote-footer'>{ author }</footer>
            </blockquote> */}
            <button 
                className='btn btn-primary'
                onClick={ increment }>
                Next
            </button>
        </>
    );
}

export default MultipleCustomHooks;