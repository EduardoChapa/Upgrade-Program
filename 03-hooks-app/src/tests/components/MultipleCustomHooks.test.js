import React from 'react';
import { shallow } from 'enzyme'
import MultipleCustomHooks from '../../components/use-fetch/MultipleCustomHooks';
import useFetch from '../../hooks/useFetch';
import useCounter from '../../hooks/useCounter';
jest.mock('../../hooks/useFetch');
jest.mock('../../hooks/useCounter');

describe('Testing MultipleCustomHooks', () => {

    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    });

    test('should display correctly', () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should display correct information', () => {
        useFetch.mockReturnValue({
            data: [{
                author: 'Ricardio',
                quote: 'Hi Hello'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);

        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('Hi Hello');
        console.log(wrapper.html());
    });

});