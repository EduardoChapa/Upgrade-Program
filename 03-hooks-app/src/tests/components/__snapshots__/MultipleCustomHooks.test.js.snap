// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MultipleCustomHooks should display correctly 1`] = `
<Fragment>
  <h1>
    Funny Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next
  </button>
</Fragment>
`;
